fujiji-server/src/controllers/auth.js

```diff
@@ -0,0 +1,87 @@
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+
+const { config } = require('../config/config');
+
+const { getUserByEmail, createUser } = require('../repositories/user');
+
+const {
+  APIError,
+  InvalidPasswordError,
+  UserNotFoundError,
+} = require('../errors');
+
+async function signUp(req, res, next) {
+  try {
+    const {
+      email, password, name, phoneNumber,
+    } = req.body;
+
+    const existingUser = await getUserByEmail(email);
+
+    if (!existingUser) {
+      const hash = await bcrypt.hash(password, 10);
+
+      const newUser = await createUser(email, hash, name, phoneNumber);
+
+      const authToken = jwt.sign(
+        {
+          id: newUser.user_id,
+          email,
+          password,
+        },
+        config.JWT_SECRET,
+        { expiresIn: '3h' },
+      );
+
+      return res.status(200).json({
+        authToken,
+        user: {
+          id: newUser.user_id,
+          name,
+          email,
+          phoneNumber,
+        },
+      });
+    }
+    return next(new APIError('User with email address already exists', 400));
```

Comment:
To be consistent, maybe make a `UserRegisteredEmailError` for this, and say  'User with *this* email address already exists' instead (just to be more clear)?

Reply:
Yeah, you are right! updated this